### PR TITLE
Merge changes to ga_locate_region function

### PR DESCRIPTION
**Pull request by bjpalmer**
This PR has some changes to GA locate region that should fix it when using block-cyclic and tiled data distributions.

### CODE REVIEW

File: armci/src/common/capi.c [hunk 1]
```diff
@@ -208,15 +208,6 @@ int ARMCI_Init_mpi_comm(MPI_Comm comm)
 #endif
 
 
-#if HAVE_SYS_WEAK_ALIAS_PRAGMA
```
Comment (jeffhammond) [inline armci/src/common/capi.c:211]: are you sure this is should be part of this commit?  it is unrelated.

File: armci/src/common/capi.c [hunk 2]
```diff
@@ -208,15 +208,6 @@ int ARMCI_Init_mpi_comm(MPI_Comm comm)
...
-#   pragma weak ARMCI_Init
```
Comment (jeffhammond) [inline armci/src/common/capi.c:212]: at the very least, we have a problem if the weak symbol doesn't match the name of the function, no?

File: armci/src/common/capi.c (hunk 1 above)
Comment (ajaypanyala) [reply armci/src/common/capi.c:211]: Commit was related to Issue #292

**Comment by bjpalmer** [comment]
This actually has some issues if you are using one of the non-standard data distributions. I need to do some further debugging on this.

**Comment by bjpalmer** [comment]
I think everything is working. I will change status from draft.

**Pull request merged**